src/ext/java/org/opentripplanner/ext/vehicleparking/hslpark/HslParkUpdater.java

```diff
@@ -0,0 +1,288 @@
+            var wheelChairAccessiblePlaces =
+                    maybeCapacity.map(c -> hasPlaces(capacity.getWheelchairAccessibleCarSpaces()))
+                            .orElse(false);
+
+            this.lastFacilitiesFetchTime = System.currentTimeMillis();
+            return VehicleParking.builder()
+                    .id(vehicleParkId)
+                    .name(name)
+                    .state(state)
+                    .x(x)
+                    .y(y)
+                    .capacity(capacity)
+                    .bicyclePlaces(bicyclePlaces)
+                    .carPlaces(carPlaces)
+                    .wheelchairAccessibleCarPlaces(wheelChairAccessiblePlaces)
+                    .tags(tags)
+                    .entrance((builder) -> builder
+                            .entranceId(
+                                    new FeedScopedId(feedId, vehicleParkId.getId() + "/entrance"))
+                            .name(name)
+                            .x(x)
+                            .y(y)
+                            .walkAccessible(true)
+                            .carAccessible(carPlaces || wheelChairAccessiblePlaces))
+                    .build();
+        }
+        catch (Exception e) {
+            log.warn("Error parsing park " + vehicleParkId, e);
+            return null;
+        }
+    }
+
+    private HslParkPatch parseUtilization(JsonNode jsonNode) {
+        var vehicleParkId = createIdForNode(jsonNode, "facilityId");
+        try {
+            String capacityType = jsonNode.path("capacityType").asText();
+            Integer spacesAvailable = parseIntegerValue(jsonNode, "spacesAvailable");
+            return new HslParkPatch(vehicleParkId, capacityType, spacesAvailable);
+        }
+        catch (Exception e) {
+            log.warn("Error parsing park utilization" + vehicleParkId, e);
+            return null;
+        }
+    }
+
+    public Integer parseIntegerValue(JsonNode jsonNode, String fieldName) {
+        if (!jsonNode.has(fieldName)) {
+            return null;
+        }
+        return jsonNode.get(fieldName).asInt();
+    }
+
+    private VehicleParkingSpaces parseVehicleSpaces(
+            JsonNode node,
+            String bicycleTag,
+            String carTag,
+            String wheelchairAccessibleCarTag
+    ) {
+        var bicycleSpaces = parseIntegerValue(node, bicycleTag);
+        var carSpaces = parseIntegerValue(node, carTag);
+        var wheelchairAccessibleCarSpaces = parseIntegerValue(node, wheelchairAccessibleCarTag);
+
+        if (bicycleSpaces == null && carSpaces == null && wheelchairAccessibleCarSpaces == null) {
+            return null;
+        }
+
+        return createVehiclePlaces(carSpaces, wheelchairAccessibleCarSpaces, bicycleSpaces);
+    }
+
+    private VehicleParkingSpaces createVehicleAvailability(List<HslParkPatch> patches) {
+        Integer carSpaces = null;
+        Integer wheelchairAccessibleCarSpaces = null;
+        Integer bicycleSpaces = null;
```

Comment:
I can remove replace these with the builder but I think we also need a boolean that tells if there are any handled spaces or else instead of the VehicleParkingSpaces instance with null spaces we can just store the availability as null.

Reply:
Refactored this a bit in https://github.com/opentripplanner/OpenTripPlanner/pull/3796/commits/8564c891621ebe8e75ddaf33ad4b7bb9562f7f6c